resources/world/world_data.gd

```diff
@@ -429,6 +477,38 @@ func get_cells_for(p_type: int) -> Array:
 	return value
 
 
+# Returns and Array of cell indexes that are already set to spawn something after world generation
+# ends, and so shouldn't be used for spawning anything else.
+func get_occupied_cells() -> Array:
+	var value := []
+	
+	if player_spawn_position != INVALID_STARTING_CELL:
+		value.append(player_spawn_position)
+	
+	value.append_array(_objects_to_spawn.keys())
+	
+	return value
+
+
+# Checks if a cell index is free
+func is_cell_free(cell_index: int) -> bool:
```

Comment:
This can be done more efficiently by checking if the _objects_to_spawn dictionary has the cell index as a key using Dictionary.has(), and checking if the cell_index is the same as the player spawn cell index.

Reply:
As far as I know `not cell_index in occupied_cells` is syntatic sugar for `has` and not walking into the array, and I remember seeing it recommended over `.has` because of some gdscript optimization but couldn't find it again, so I migth be remembering wrong. 

I'll change this as requested, but I though the benefit of this is that we already have two different conditions for checking if cells are free, and there will possibly be more later, as I think characters should have a different property than `_objects_to_spawn` and later on, with objects that spawn more than one cell, this might have to change again. 

So I was thinking it's easier to read this code and understand it, and we can just keep changing `get_occupied_cells` as the rules change. Plus other places might want to get all occupied cells for other reasons.

Reply:
Array.has() in an array walks the array, The array has no guarantees of any order, so it can't do anything efficient like a binary search, so it walks the array until it finds a match.
In the godot source, Array.find() calls Vector.find() inside an internal ArrayPrivate (https://github.com/godotengine/godot/blob/3.5/core/array.cpp#L210)
Which calls CowData.find() (https://github.com/godotengine/godot/blob/3.5/core/vector.h#L90)
Which walks the array (https://github.com/godotengine/godot/blob/3.5/core/cowdata.h#L336)